modules/tensor_mechanics/src/userobjects/StepUserObject.C

```diff
@@ -0,0 +1,96 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "StepUserObject.h"
+#include <limits>
+
+registerMooseObject("TensorMechanicsApp", StepUserObject);
+
+InputParameters
+StepUserObject::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+
+  params.addRequiredParam<std::vector<Real>>(
+      "times",
+      "The times at which the objects are to be enabled/disabled at the beginning of the steps.");
```

Comment:
These are the start times for each step?

Reply:
I am making the guess that you are suggesting to have a more elaborate name, such as `step_start_times`?